plugin/storage/integration/elasticsearch_test.go

```diff
@@ -192,3 +195,49 @@ func (s *StorageIntegration) testArchiveTrace(t *testing.T) {
 		CompareTraces(t, &model.Trace{Spans: []*model.Span{expected}}, actual)
 	}
 }
+
+func (s *StorageIntegration) testLargeTraceSearchAfter(t *testing.T) {
+	defer s.cleanUp(t)
+
+	var expected []*model.Span
+	p := model.NewProcess("large_trace", model.KeyValues{})
+	now := time.Now()
+	tID := model.NewTraceID(rand.Uint64(), rand.Uint64())
+	parent := &model.Span{
+		OperationName: "parent",
+		StartTime:     now,
+		TraceID:       tID,
+		SpanID:        model.NewSpanID(rand.Uint64()),
+		References:    []model.SpanRef{},
+		Process:       p,
+	}
+	expected = append(expected, parent)
+
+	// Report large number of spans per trace to trigger search_after/pagination
+	// Use the same timestamp to test sorting works
+	for i := 0; i < 11000; i++ {
+		child := &model.Span{
+			OperationName: "child",
+			StartTime:     now,
+			TraceID:       tID,
+			SpanID:        model.NewSpanID(rand.Uint64()),
+			References:    []model.SpanRef{{TraceID: parent.TraceID, SpanID: parent.SpanID}},
+			Process:       p,
+		}
+		require.NoError(t, s.SpanWriter.WriteSpan(child))
+		expected = append(expected, parent)
+	}
+
+	require.NoError(t, s.SpanWriter.WriteSpan(parent))
+	s.refresh(t)
+
+	var actual *model.Trace
+	found := s.waitForCondition(t, func(t *testing.T) bool {
+		var err error
+		actual, err = s.SpanReader.GetTrace(context.Background(), tID)
+		return err == nil && len(actual.Spans) == len(expected)
+	})
+	if !assert.True(t, found) {
+		CompareTraces(t, &model.Trace{Spans: expected}, actual)
```

Comment:
This might still not be true because its going to sort based on ascending order of `spanID` as well.

Reply:
It seems that tests are pasing

Reply:
Travis doesn't run integration tests for elasticsearch, right? Just `STORAGE=memory`

Reply:
Travis runs integration tests for all storge backeds (not sure about badger). For elasticsearch we run it againt multiple ES versions - 5.x, 6.x and 7.x.

Reply:
Aha, my bad, was blinding out the ES_INTEGRATION_TEST test because I so rarely see it